shell/browser/net/proxying_url_loader_factory.cc

```diff
@@ -243,6 +243,26 @@ void ProxyingURLLoaderFactory::InProgressRequest::OnReceiveResponse(
     network::mojom::URLResponseHeadPtr head,
     mojo::ScopedDataPipeConsumerHandle body,
     absl::optional<mojo_base::BigBuffer> cached_metadata) {
+  // Chromium sets the kContentType and kLastModified fields in the header
+  // always.
+  if (request_.url.SchemeIs(url::kFileScheme) &&
+      !request_.url.ExtractFileName().empty() && head->headers) {
+    if (request_.url.ExtractFileName().find(".") == std::string::npos &&
+        head->headers->HasHeaderValue(net::HttpRequestHeaders::kContentType,
+                                      "text/plain")) {
+      // Having content-type set causes an issue for css style sheets with no
+      // file extension (asset pool files). In this case the content type is set
+      // to text/plain as the mime sniffer doesn't support detecting a css file,
+      // which causes the style sheet to be ignored. To fix this remove the
+      // content type.
+      head->headers->RemoveHeader(net::HttpRequestHeaders::kContentType);
+    }
+    if (head->headers->HasHeader(net::HttpResponseHeaders::kLastModified)) {
+      // We don't want to cache files so remove the last modified header.
+      head->headers->RemoveHeader(net::HttpResponseHeaders::kLastModified);
+    }
+  }
```

Comment:
Can't we have a patch which reverts the patch on Chromium, rather than removing them back here?

Reply:
Do you mean revert all 16 files of the chromium patch or just file_url_loader_factory.cc?
I felt like from a maintenance point of view patching a single Electron file would be better than reverting a chromium patch which may also cause conflicts that need to be resolved when we update Chromium versions in future?

Reply:
Not a complete revert but we can revert the following lines:
```
-    if (head->headers) {
-      head->headers->AddHeader(net::HttpRequestHeaders::kContentType,
-                               head->mime_type);
+    if (!head->headers) {
+      head->headers =
+          base::MakeRefCounted<net::HttpResponseHeaders>("HTTP/1.1 200 OK");
     }
+    head->headers->AddHeader(net::HttpRequestHeaders::kContentType,
+                             head->mime_type);
+    // We add a Last-Modified header to file responses so that our
+    // implementation of document.lastModified can access it (crbug.com/875299).
+    head->headers->AddHeader(net::HttpResponseHeaders::kLastModified,
+                             base::TimeFormatHTTP(info.last_modified));
```

from 
```
commit 3bf8acaeb418b2a0832c4b87750fddcbdec5715d
Author: Ryan Tarpine <rtarpine@chromium.org>
Date:   Thu Jan 20 02:43:16 2022 +0000
```
I first thought this would cause Chromium tests to fail, but that doesn't matter. Because we don't run the Chromium tests in Electron tests. This is going to be .patch file anyway.

Reply:
I tried reverting that part of the patch. However, this is not enough and a further patch will need to be made. This is because when running on QtWebEngine head->headers is null, but, when run on Electron it's not. Field head->headers  is initialised to extra_response_headers. I can't see an obvious reason for the difference in behaviour (i.e. there is no patch in QtWebEngine that does this). This means some behaviour change elsewhere in the file loading sequence has changed and we now have a header set.
Based on this I think the change I have made is safest, rather than trying to find and patch other bits of Chromium code.